scripts/tests/upgrade-test.sh

```diff
@@ -72,9 +70,19 @@ for upgrade_path in "${upgrade_paths[@]}"; do
       fi
     done
 
-    upgrade_tests+=(
-      "fm-run-test fedimintd-${versions_str} devimint upgrade-tests fedimintd --paths $(printf "%s " "${fedimintd_paths[@]}")"
-    )
+    # If the first version is before LNv2's stablization, then we run the upgrade tests with LNv2 off since that is how users will upgrade in production.
+    # If it is equal or greater than LNv2 stablization, then we need to run the upgrade tests with LNv2 on and off.
+    if version_lt $first_version $LNV2_STABLE_VERSION; then
```

Comment:
[nit] I think we can simplify to only one `version_lt` check then loop over the array of `lnv2_flags` when we append to `upgrade_tests`

```bash
if version_lt "$first_version" "$LNV2_STABLE_VERSION"; then
  lnv2_flags=(0)
else
  lnv2_flags=(0 1)
fi

...

for enable_lnv2 in "${lnv2_flags[@]}"; do
  upgrade_tests+=(
    "fm-run-test fedimintd-${versions_str}-lnv2-{$enable_lnv2} devimint upgrade-tests --lnv2 $enable_lnv2 fedimintd --paths $(printf "%s " "${fedimintd_paths[@]}")"
  )
done
...
```